acstools/polarization_tools.py

```diff
@@ -0,0 +1,205 @@
+
+        self.hrc_efficiency = Table(self.table_data['efficiency']['hrc'],
+                                    names=('filter', 'pol0', 'pol60', 'pol120'))
+
+
+class Polarization:
+    """
+    Class for handling ACS polarization data.
+
+    Examples
+    --------
+
+    >>> from acstools.polarization_tools import Polarization
+    >>> vela_181 = Polarization(63684, 67420, 63752, 'F606W', 'WFC', 348.084)
+    >>> vela_181.calc_stokes()
+    >>> vela_181.calc_polarization()
+    >>> print(f'I = {vela_181.stokes_i:.2f}, Q = {vela_181.stokes_q:.2f}, U = {vela_181.stokes_u:.2f}')
+    >>> print(f'Polarization: {vela_181.polarization * 100:.2f} %, Angle: {vela_181.angle:.2f} deg')
+    """
+
+    def __init__(self, pol0, pol60, pol120, filter_name, detector, pav3):
+        """
+        Parameters
+        ----------
+        pol0 : float
+            Photometric measurement in POL0 filter. Units: electrons or electrons/second.
+        pol60 : float
+            Photometric measurement in POL60 filter. Units: electrons or electrons/second.
+        pol120 : float
+            Photometric measurement in POL120 filter. Units: electrons or electrons/second.
+        filter_name : str
+            Name of the filter crossed with the polarization filter, e.g., F606W.
+        detector : str
+            Name of the ACS detector used for the observation. Must be either WFC or HRC.
+        pav3 : float
+            Position angle of the HST V3 axis. This is stored in the ACS primary header under
+            keyword PA_V3. Units: degrees.
+        """
+
+        self.pol0 = pol0
+        self.pol60 = pol60
+        self.pol120 = pol120
+
+        self.filter_name = filter_name
+        self.detector = detector
+        self.pav3 = pav3
+
+        # Check if detector is a valid value.
+        if self.detector.lower() not in ['wfc', 'hrc']:
+            raise ValueError('Detector must be either WFC or HRC')
+
+        self.stokes_i = None
+        self.stokes_q = None
+        self.stokes_u = None
+
+        self.polarization = None
+        self.angle = None
+
+        # Get correction terms that we need from the polarization tables.
+        tables = PolarizerTables()
+        leak_tab = Table(tables.table_data['transmission'][detector.lower()])
+        self.transmission_correction = leak_tab[np.where(leak_tab['filter'] == filter_name.upper())]['correction'][0]
+
+        eff_tab = Table(tables.table_data['efficiency'][detector.lower()])
+        self.c0 = eff_tab[np.where(eff_tab['filter'] == filter_name.upper())]['pol0'][0]
+        self.c60 = eff_tab[np.where(eff_tab['filter'] == filter_name.upper())]['pol60'][0]
+        self.c120 = eff_tab[np.where(eff_tab['filter'] == filter_name.upper())]['pol120'][0]
+
+    def calc_stokes(self):
+        """
+        Calculate Stokes parameters using attributes set at initialization.
+        """
+
+        self.stokes_i, self.stokes_q, self.stokes_u = self._calc_stokes([self.pol0, self.pol60, self.pol120],
```

Comment:
Is there a reason why this calls a hidden method that isn't used anywhere else, instead of moving the code from `_calc_stokes` here and get rid of the hidden method?